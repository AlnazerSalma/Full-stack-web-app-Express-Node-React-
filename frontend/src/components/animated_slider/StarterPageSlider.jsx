import React, { useRef } from "react";
import useSliderAnimation from "../../hooks/useSliderAnimation";
import "./AnimatedSlider.css";
import mediaRenderer from "../../utils/mediaRenderer";

const StarterPageSlider = ({ items }) => {
  const sliderTrackRef = useRef(null);
  useSliderAnimation(sliderTrackRef, items, 40);
  return (
    <div className="slider-wrapper">
      
      <div className="slider-track" ref={sliderTrackRef}>
        {items.map((item, index) => (
          <div className="slider-item" key={index}>
              <div className="slider-content">
              <h2>{item.name}</h2>
              <p>{item.desc}</p>
            </div>
           {mediaRenderer({
              src: item.image,
              type: item.type,
              alt: item.name,
              className: "media-class-name",

            })}
          </div>
        ))}
      </div>
    </div>
  );
};

export default StarterPageSlider;
