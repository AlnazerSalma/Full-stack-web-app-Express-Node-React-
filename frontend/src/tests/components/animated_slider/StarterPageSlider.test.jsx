import React from 'react';
import { render, screen } from '@testing-library/react';
import StarterPageSlider from '../../../components/animated_slider/StarterPageSlider'; // adjust path as needed
import mockMediaRenderer from '../../__test_utils__/mockMediaRenderer'; 
import mockStartPageSlide from '../../__mocks_data__/mockStartPageSlide'; 

jest.mock("../../../utils/mediaRenderer", () => ({
  __esModule: true,
  default: require("../../__test_utils__/mockMediaRenderer").default,
}));


jest.mock('../../../hooks/useSliderAnimation', () => jest.fn());

describe('StarterPageSlider', () => {

  it('renders all slider items with correct content', () => {
    render(<StarterPageSlider items={mockStartPageSlide} />);

    // Check all item names and descriptions render
    mockStartPageSlide.forEach(({ name, desc }) => {
      expect(screen.getByText(name)).toBeInTheDocument();
      expect(screen.getByText(desc)).toBeInTheDocument();
    });

    expect(mockMediaRenderer).toHaveBeenCalledTimes(mockStartPageSlide.length);

    // Check mediaRenderer was called with correct props for each item
   mockStartPageSlide.forEach((item, index) => {
  expect(mockMediaRenderer.mock.calls[index][0]).toEqual(
    expect.objectContaining({
      src: item.image,
      type: item.type,
      alt: item.name,
      className: "media-class-name",
    })
  );
});

    // Check that the correct number of slider-item rendered
    const sliderItems = screen.getAllByRole('heading', { level: 2 });
    expect(sliderItems.length).toBe(mockStartPageSlide.length);
  });

  it('matches snapshot', () => {
  const { asFragment } = render(<StarterPageSlider items={mockStartPageSlide} />);
  expect(asFragment()).toMatchSnapshot();
});

});
