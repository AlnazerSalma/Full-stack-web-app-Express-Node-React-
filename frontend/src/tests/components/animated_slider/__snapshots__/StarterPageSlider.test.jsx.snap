// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarterPageSlider matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="slider-wrapper"
  >
    <div
      class="slider-track"
    >
      <div
        class="slider-item"
      >
        <div
          class="slider-content"
        >
          <h2>
            Slide 1
          </h2>
          <p>
            Description 1
          </p>
        </div>
      </div>
      <div
        class="slider-item"
      >
        <div
          class="slider-content"
        >
          <h2>
            Slide 2
          </h2>
          <p>
            Description 2
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
